lib/internal/modules/cjs/loader.js

```diff
@@ -654,15 +617,23 @@ Module._findPath = function(request, paths, isMain) {
   let trailingSlash = request.length > 0 &&
     request.charCodeAt(request.length - 1) === CHAR_FORWARD_SLASH;
   if (!trailingSlash) {
-    trailingSlash = /(?:^|\/)\.?\.$/.test(request);
+    trailingSlash = trailingSlashRegex.test(request);
```

Comment:
```suggestion
    trailingSlash = RegExpPrototypeTest(trailingSlashRegex, request);
```
(plus the corresponding primordial import)

Reply:
Added.